eslint.config.mjs

```diff
@@ -1,280 +1,221 @@
+      'require-await': 'error',
+      'sort-imports': ['error', { ignoreDeclarationSort: true }]
+    }
+  })
 
-      'jsdoc/require-jsdoc': 'off',
-      'jsdoc/require-param': 'off',
-      'jsdoc/require-returns': 'off',
-      'jsdoc/require-param-type': 'off',
-      'import/order': [
-        'error',
-        {
-          pathGroups: [
-            {
-              pattern: '#vue-router',
-              group: 'external'
-            }
-          ]
-        }
-      ],
-      'import/no-restricted-paths': [
-        'error',
-        {
-          zones: [
-            {
-              from: 'packages/nuxt/src/!(core)/**/*',
-              target: 'packages/nuxt/src/core',
-              message: 'core should not directly import from modules.'
-            },
-            {
-              from: 'packages/nuxt/src/!(app)/**/*',
-              target: 'packages/nuxt/src/app',
-              message: 'app should not directly import from modules.'
-            },
-            {
-              from: 'packages/nuxt/src/app/**/index.ts',
-              target: 'packages/nuxt/src',
-              message: 'should not import from barrel/index files'
-            },
-            {
-              from: 'packages/nitro',
-              target: 'packages/!(nitro)/**/*',
-              message: 'nitro should not directly import other packages.'
-            }
-          ]
-        }
-      ],
-      '@typescript-eslint/consistent-type-imports': [
-        'error',
-        {
-          disallowTypeAnnotations: false
-        }
-      ],
+  .override('nuxt/typescript/rules', {
+    rules: {
       '@typescript-eslint/ban-ts-comment': [
         'error',
         {
           'ts-expect-error': 'allow-with-description',
           'ts-ignore': true
         }
       ],
-      '@typescript-eslint/prefer-ts-expect-error': 'error',
+      '@typescript-eslint/no-dynamic-delete': 'off',
       '@typescript-eslint/no-unused-vars': [
         'error',
         {
           argsIgnorePattern: '^_',
-          varsIgnorePattern: '^_',
-          ignoreRestSiblings: true
+          ignoreRestSiblings: true,
+          varsIgnorePattern: '^_'
         }
       ],
-      'jsdoc/check-tag-names': [
-        'error',
-        {
-          definedTags: ['__NO_SIDE_EFFECTS__']
-        }
-      ]
-    },
-    settings: {
-      jsdoc: {
-        ignoreInternal: true,
-        tagNamePreference: {
-          warning: 'warning',
-          note: 'note'
-        }
+      '@typescript-eslint/triple-slash-reference': 'off',
+      '@typescript-eslint/unified-signatures': 'off',
+      ...{
+        // TODO: Discuss if we want to enable this
+        '@typescript-eslint/ban-types': 'off',
+        // TODO: Discuss if we want to enable this
+        '@typescript-eslint/no-explicit-any': 'off',
+        // TODO: Discuss if we want to enable this
+        '@typescript-eslint/no-invalid-void-type': 'off'
```

Comment:
These are the rules that creating quite a few warnings - we might want to enable them eventually but would requires some updates to the code.